### DISCUSSION TITLE
How to exclude the specific file from analysis?

### DISCUSSION THREAD

**Discussion by vplagov** [JetBrains/Qodana#92] [Q&A] [answered]
Is it possible to exclude a specific Java/Kotlin class or file from the analysis? I don't want to exclude the whole package because there are other files under this package that I want to be analysed. I want to exclude just a specific class/file.

A bit more details about my case.
I have an enum class with 31 enums. None of them is called directly. 
```kotlin
enum class FooBar(val value: String) {
    FOO("FooValue"),
    BAR("BarValue"),
}
```
I'm mapping these enums to the response of the BE service:
```kotlin
fun getAvailableFooBar(): List<FooBar> = client.getFooBar().responseBody()
```

Then I'm using it as follows
```kotlin
val fooBarValues = service.getAvailableFooBar().map { it.value }
```

So, I'm not using enums individually, but using the `value` field of those enums.
During the analysis, Qodana says that these enums are not used and can be deleted. If I delete them, then the last code block won't work.

So, can I exclude this particular enum class from the analysis? Or maybe I'm using enums here the wrong way?

**Comment by brichbash** [comment, marked as answer]
Hello,

You can exclude files using `qodana.yaml`:
```
version: "1.0"
exclude:
  - name: All
    paths:
      - relative/path/to/your/file.kt
```
And about the enums themselves, your issue looks like a Kotlin Plugin issue. Is it reproducible when using in your IDE?

**Reply by vplagov** [reply]
Sorry for the long reply.
Thanks for the help. That works!

**Reply by pesterevvladislav** [reply]
Nope, not working for me

**Reply by brichbash** [reply]
@pesterevvladislav hey, could you please share some details? What is your current setup and what you want to achieve? You can also contact us at qodana-support@jetbrains.com

**Reply by pesterevvladislav** [reply]
@brichbash I was trying to exclude a folder from indexing, this answer doesn't completely answer the question.

**Reply by brichbash** [reply]
@pesterevvladislav  The initial question was about analysis scope which is different from indexing. 
The mentioned `exclude` option indeed doesn't affect whether the folders are indexed or not. However, if you use Qodana for JVM, you can set which Gradle/Maven modules to import which does affect if they are indexed or not. Please check [this documentation section](https://www.jetbrains.com/help/qodana/qodana-yaml.html#Configure+Java+and+Kotlin+projects+in+monorepo).

If this doesn't meet your needs, please share more details about your use case, and I'll file a feature request if it isn't supported

**Comment by juliocebrito** [comment]
Hello,

How I can exclude all migrations directories of migrations when de project has 50 apps, without relative paths

version: "1.0"
profile:
  name: qodana.recommended
exclude:
  - name: All
    paths:

        - app1/migrations
        - app2/migrations
        ...
        - app50/migrations

Thanks in advance.

**Reply by brichbash** [reply]
Hello,
You can try out [our new YAML profile feature](https://www.jetbrains.com/help/qodana/custom-profiles.html).
All you need to do is to create `<your-filename>.yaml`:
```
name: "Excluded migrations"
baseProfile: qodana.recommended
inspections:
  - group: ALL
    ignore:
      - "scope#file:**/migrations//*"
```
And then set the profile in Qodana [by path.](https://www.jetbrains.com/help/qodana/inspection-profiles.html#Profile+path)
If you have any questions, please let me know.
